### PR TITLE
Enable gpload to switch to gpdb5 text format if enable_custom_format guc is set.

### PR DESCRIPTION
**Pull request by zhaorui20005**
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change # Because the text_in custom formatter is in gpfmt_gpss.so, and the test cases can only to be added in gpss repo, here because the switching action is controlled by gpload_enable_custom_format guc, so we just make sure all the old test cases passed.
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by zhaorui20005** [comment]
This pr adds one guc called 'gpload_enable_custom_format' to let users enable gpload using custom formatter 'text_in' written before to create external table in gpdb5 text format. Because as the postgres codes updates, the gpdb6 text format acts in a different way in some details(delimiter=off , or '\.' will be an end of line) from gpdb5 text format. This will make users failed to load data into gpdb. If users does not define SQL:BEFORE 'set gpload_enable_custom_format', they will still use gpload in the old ways,

File: src/backend/utils/misc/guc_gp.c [hunk 1]
```diff
@@ -3119,6 +3119,17 @@ struct config_int ConfigureNamesInt_gp[] =
...
 	},
 
+	{
+		{"gpload_enable_custom_format", PGC_USERSET, EXTERNAL_TABLES,
+			gettext_noop("Whether or not to enable switch 'text' format to custom formatter 'text_in' to simulate actions in gpdb5."),
+			gettext_noop("Default value is 0."),
+			GUC_NOT_IN_SAMPLE
+		},
+		&gpload_enable_custom_format,
+		0, 0, INT_MAX,
```
Comment (huiliang-liu) [inline src/backend/utils/misc/guc_gp.c:3129]: use bool type is better

File: gpMgmt/bin/gpload_test/gpload2/query37.ans [hunk 2]
```diff
@@ -3,7 +3,7 @@
...
 2018-11-05 22:52:11|INFO|did not find a staging table to reuse. creating staging_gpload_reusable_afbaac0da7ced19791c9ab9c537f41d3
 2018-11-05 22:52:11|INFO|did not find an external table to reuse. creating ext_gpload_reusable_601e34fe_e10a_11e8_b2e8_00505698a2d7
-2018-11-05 22:52:11|ERROR|could not run SQL "create external table ext_gpload_reusable_601e34fe_e10a_11e8_b2e8_00505698a2d7("s1" text,"s2" text,"s3" text,"dt" timestamp without time zone,"n1" smallint,"n2" integer,"n3" bigint,"n4" numeric,"n5" numeric,"n6" real,"n7" double precision)location('gpfdist://127.0.0.1:8081//home/gpadmin/workspace/gpdb/gpMgmt/bin/gpload_test/gpload2/data_file.txt') format'text' (delimiter '|' null '\N' escape '\') encoding'xxxx' ": ERROR:  xxxx is not a valid encoding name
+2018-11-05 22:52:11|ERROR|could not run SQL "create external table ext_gpload_reusable_601e34fe_e10a_11e8_b2e8_00505698a2d7("s1" text,"s2" text,"s3" text,"dt" timestamp without time zone,"n1" smallint,"n2" integer,"n3" bigint,"n4" numeric,"n5" numeric,"n6" real,"n7" double precision)location('LOCATION') format'text' (delimiter '|' null '\N' escape '\') encoding'xxxx' ": ERROR:  xxxx is not a valid encoding name
```
Comment (huiliang-liu) [inline gpMgmt/bin/gpload_test/gpload2/query37.ans:6]: why case 37 fails by this PR? and I think init_file can handle the location in init_file.

File: src/backend/utils/misc/guc_gp.c (hunk 1 above)
Comment (zhaorui20005) [reply src/backend/utils/misc/guc_gp.c:3129]: All right. We are going to move the guc into gpss.so later and modify the gpload after that.

File: gpMgmt/bin/gpload_test/gpload2/query37.ans (hunk 2 above)
Comment (zhaorui20005) [reply gpMgmt/bin/gpload_test/gpload2/query37.ans:6]: In my env, the root path fails match the origin path '/home/gpadmin/workspace', so it fails.

File: gpMgmt/bin/gpload.py [hunk 3]
```diff
@@ -1905,6 +1905,36 @@ def read_columns(self):
...
+    def test_enable_custom_format(self):
+        # Test custom format guc
+        self.enable_custom_format = 0;
+        queryString = """show gpload_enable_custom_format;"""
+        resultList = self.db.query(queryString.encode('utf-8')).getresult()
+        val = int(resultList[0][0])
+        if val != 0:
+            self.enable_custom_format = 1
+
+    def test_custom_formatter(self):
```
Comment (huiliang-liu) [inline gpMgmt/bin/gpload.py:1917]: rename the two function as check_custom_formatter()/check_enable_custom_format()

Comment (zhaorui20005) [reply gpMgmt/bin/gpload.py:1917]: All right, the test prefix seems to be used in testing cases.

File: gpMgmt/bin/gpload_test/gpload2/query37.ans (hunk 2 above)
Comment (zhaorui20005) [reply gpMgmt/bin/gpload_test/gpload2/query37.ans]: Fixed.

**Review by pf-qiu — APPROVED** [review]

**Pull request merged**